src/Scanner.php

```diff
@@ -0,0 +1,267 @@
+            // Replace '/' with '__' in column name.
+            $column = str_replace('/', '__', $column);
+            $field_info['source'][$fieldkey] = $column;
+          }
+          else {
+            $field_info['source'][] = $column;
+          }
+
+          // Check if we have such field in this entity.
+          if (isset($instances[$fieldname])) {
+            $field_info['field_type'] = $instances[$fieldname]->getType();
+            $field_info['field_storage_definition'] = $instances[$fieldname]->getFieldStorageDefinition();
+            $reference_target_type = $target_bundles = NULL;
+            // Get target info for field types.
+            switch ($field_info['field_type']) {
+              case 'image':
+              case 'file':
+                $reference_target_type = 'file';
+                $target_bundles = ['file'];
+                break;
+
+              case 'entity_reference':
+              case 'entity_reference_revisions':
+                $reference_definition = $instances[$fieldname]->getItemDefinition()->getSettings();
+                $reference_target_type = $reference_definition['target_type'];
+                if (!empty($reference_definition['handler_settings']['target_bundles'])) {
+                  $target_bundles = $reference_definition['handler_settings']['target_bundles'];
+                }
+                else {
+                  // Get all bundles for referenced target type.
+                  $target_bundles = array_keys($this->entityTypeBundleInfo->getBundleInfo($reference_target_type));
+                }
+                break;
+            }
+            if ($reference_target_type && $target_bundles) {
+              $field_info['is_reference'] = TRUE;
+              $field_info['reference_target_type'] = $reference_target_type;
+              $field_info['target_bundles'] = $target_bundles;
+
+              // Check if we have sources for referenced entity type.
+              if (isset($sources[$reference_target_type])) {
+                foreach ($target_bundles as $target_bundle) {
+                  // Check if we have sources for referenced bundle.
+                  if (isset($sources[$reference_target_type][$target_bundle])
+                    && !empty($sources[$reference_target_type][$target_bundle]['migration_id'])) {
+                    // If have source then add dependency.
+                    $related_migration = $sources[$reference_target_type][$target_bundle]['migration_id'];
+                    $field_info['dependencies'][] = $related_migration;
+                  }
+                }
+              }
+            }
+
+            if ($instances[$fieldname]->getFieldStorageDefinition()->getCardinality() != 1) {
+              $field_info['multiple'] = TRUE;
+            }
+
+            $fields_info[$fieldname] = $field_info;
+          }
+          else {
+            $this->logger->warning(
+              'Field %field not found for %entity %bundle.',
```

Comment:
I suggest wording update to `'Field %field not found for %bundle %entity.',`

Reply:
done